### PR TITLE
CLI

### PR DESCRIPTION
**Pull request by victor-hugo-dc**
**Command-Line Interface Tool** 
From #104, I took the liberty to design a simple CLI tool that handles a variety of file type inputs (not just .csv) and can either take in a users token in the command line or can pull from the `.env` file directly. For the best user experience, I have set the default output of pandasai to be verbose and made sure to implement various error checks that print out only a single line detailing what went wrong.

Screenshot 2023-05-10 at 7 45 59 PM

### CODE REVIEW

File: pai/__main__.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+from pandasai.llm.open_assistant import OpenAssistant
+import pandas as pd
+
+import os
+from dotenv import load_dotenv
+
+dotenv_path = os.path.join(os.path.dirname(__file__), '..', '.env')
+load_dotenv(dotenv_path=dotenv_path)
+
+parser = argparse.ArgumentParser(description='Pandas AI command-line tool.')
```
Comment (avelino) [inline pai/__main__.py:13]: in the case of cli we could use the [click package](https://click.palletsprojects.com/en/8.1.x/) to simplify the implementation and maintenance - the code is more organized

**Review by avelino — CHANGES_REQUESTED** [review]
given that we are making a cli we need to think about package distribution for macOS, Linux and Windows, given the option for people to install in a "simple" way - not only via pip, e.g. homebrew for macOS

File: pai/__main__.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    except:
+        print(f"Could not find {dataset}")
+        return
+
+    if model == "openai":
+        llm = OpenAI(api_token = token or os.environ.get("OPENAI_API_KEY"))
+
+    elif model == "open-assistant":
+        llm = OpenAssistant(api_token = token or os.environ.get("HUGGINGFACE_API_KEY"))
+
```
Comment (gventuri) [inline pai/__main__.py:40]: Let's also add `Starcoder`, which we are now supporting!

File: pyproject.toml [hunk 3]
```diff
@@ -15,6 +15,7 @@ astor = "^0.8.1"
 openai = "^0.27.5"
 ipython = "^8.13.1"
 matplotlib = "^3.7.1"
+click = "^8.1.3"
```
Comment (gventuri) [inline pyproject.toml:18]: Isn't it a dev dependency?

**Review by gventuri — CHANGES_REQUESTED** [review]
Great job @victor-hugo-dc! 🎉
I'll also give it a try and let you know how it feels!

File: pyproject.toml (hunk 3 above)
Comment (victor-hugo-dc) [reply pyproject.toml:18]: I'm still fairly new to contributing to open source projects, I'm unsure if its a dev dependency; should i move it from dependencies to dev dependencies?

**Comment by HenriqueAJNB** [comment]
I agree with you in parts @avelino. What you suggested is important, but first the package must have the CLI implemented.

Your proposal is a nice to have in future versions. Perhaps you could even open another issue to track it.

**Comment by avelino** [comment]
> I agree with @avelino. What he suggested is important, but first the package must have the CLI implemented.
> 
> His proposal is a nice to have in future versions. Perhaps @avelino could open another issue to track it.

@HenriqueAJNB
"could use": **recommendation** to use, not ~~forcing~~

**Comment by gventuri** [comment]
@victor-hugo-dc tested, works super cool.

2 things left:
- fix the `linter` errors
- add an example of how to set it up and how to run it in the `README.md`

Great job @victor-hugo-dc 🎊

**Pull request merged**